### PR TITLE
Update design.md

### PR DESCRIPTION
**Pull request by frmrgr**
Mention `_` in https://dart.dev/guides/language/effective-dart/design#do-annotate-with-dynamic-instead-of-letting-inference-fail

### CODE REVIEW

File: src/_guides/language/effective-dart/design.md [hunk 1]
```diff
@@ -1518,6 +1518,8 @@ mergeJson(original, changes) => ...
 
 Note that it's OK to omit the type when Dart *successfully* infers `dynamic`.
 
+An exception to this is skipping the type annotation on an unused symbol (`_`), such as on an argument in a lambda callback.
```
Comment (sfshaza2) [inline src/_guides/language/effective-dart/design.md:1521]:
```suggestion
An exception to this is skipping the type annotation on an unused symbol (`_`),
such as on an argument in a lambda callback.
```

**Review by sfshaza2 — COMMENTED** [review]
I, too, defer to @munificent here, but I did want to break that overly long line into 2 lines.

**Comment by munificent** [comment]
It's good to call this out as an exception. Instead of "argument on a lambda callback", I would just say "an unused parameter".

File: src/_guides/language/effective-dart/design.md [hunk 2]
```diff
@@ -1548,6 +1548,7 @@ code did not specify one.
   safety and performance of static types.
 {{site.alert.end}}
 
+**Exception**: type annotation on an unused parameter (`_`) can be skipped.
```
Comment (parlough) [inline src/_guides/language/effective-dart/design.md:1551]:
```suggestion
**Exception**: Type annotations on unused parameters (`_`) can be omitted.
```

**Comment by frmrgr** [comment]
Rephrased and moved per suggestins.

**Review by parlough — APPROVED** [review]
Thanks so much for being catching this, being patient, and adjusting these changes. Looks great!

**Pull request merged**